### DISCUSSION TITLE
Allow Feed URL to be parsed with Craft::parseEnv()

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [engram-design/FeedMe#1021] [Ideas]
### Description

There might be some use cases where the Feed URL field could benefit from being environment variable aware. Here's a couple of use cases:

1. You have a staging and live environment which use a different feed endpoint per environment and you want to be able to set this without having to modify the feed properties or use a module and call a custom event i.e. `EVENT_BEFORE_FETCH_FEED`

2. A feed that may require a sensitive token, which you don't want in plain text and instead as a env var e.g.

`http://examplesite.com/feed/data.json?token=$MY_SECRET_TOKEN`

Currently you would have to use a module with a custom event to modify the feedUrl, as well as manually parse any env var to pass to Feed Me.

**Comment by jamesmacwhite** [comment]
Aliases are supported, but not env var's I don't think.

**Comment by zizther** [comment]
@angrybrad Any thoughts on this, can we use ENV variables for feed URLs?

**Reply by jamesmacwhite** [reply]
A workaround is using aliases, these are parsed in the Feed URL. This can work for having a dynamic base URL for example.

I can see environment variables being useful for both full URL or tokens/partial data though. The only way to do it currently is using Feed Me events on the before fetch feed event.

**Reply by zizther** [reply]
@jamesmacwhite I have ended up using an alias which has a ENV variable to allow easier management on the feed URL.
One extra little step which isn't really an issue and does work.

**Reply by jamesmacwhite** [reply]
> @jamesmacwhite I have ended up using an alias which has a ENV variable to allow easier management on the feed URL. One extra little step which isn't really an issue and does work.

That's what I also do to achieve a similar result
